packages/esm-patient-medications-app/src/order-basket/medication-order-form.component.tsx

```diff
@@ -232,55 +220,19 @@ export default function MedicationOrderForm({ initialOrderBasketItem, onSign, on
             <>
               <Grid className={styles.gridRow}>
                 <Column md={4}>
-                  <ComboBox
+                  <NumberInput
                     id="doseSelection"
                     light={isTablet}
-                    items={dosageOptions}
-                    selectedItem={
-                      dosageOptions?.length
-                        ? {
-                            id: orderBasketItem.dosage ? `${orderBasketItem.dosage?.value}` : null,
-                            text: orderBasketItem.dosage ? `${orderBasketItem.dosage?.value}` : null,
-                          }
-                        : null
-                    }
-                    // @ts-ignore
                     placeholder={t('editDoseComboBoxPlaceholder', 'Dose')}
-                    titleText={t('editDoseComboBoxTitle', 'Enter Dose')}
-                    itemToString={(item) => item?.text}
-                    onChange={({ selectedItem }) => {
-                      if (selectedItem) {
-                        selectedItem.id = selectedItem.id == 'draft' ? selectedItem.text : selectedItem.text;
-                        setOrderBasketItem({
-                          ...orderBasketItem,
-                          dosage: { value: Number(selectedItem.text) },
-                        });
-                      } else {
-                        setOrderBasketItem({
-                          ...orderBasketItem,
-                          dosage: { value: 0 },
-                        });
-                      }
-                      // cleaup
-                      setDosageOptions(dosageOptions.filter((opt) => opt.id != 'draft'));
-                    }}
-                    onInputChange={(value) => {
-                      if (value == 'undefined') {
-                        return;
-                      }
-                      const valueExists = value ? dosageOptions.some((opt) => `${opt.text}` == value) : null;
-                      const draftIndex = dosageOptions.findIndex((opt) => opt.id == 'draft');
-                      // validate and clean up
-                      if (draftIndex >= 0 && value) {
-                        dosageOptions[draftIndex].text = value;
-                      } else if (draftIndex >= 0) {
-                        dosageOptions.pop();
-                      }
-                      if (value && !valueExists && draftIndex == -1 && !isNaN(Number(value))) {
-                        dosageOptions.push({ id: 'draft', text: value });
-                      }
```

Comment:
@hadijahkyampeire the whole above approach of getting the dosage input and then saving the value in the object is replaced with a single line of code as is mentioned below.
```
dosage: value ? parseFloat(value) : 0
```

Also, there's no point of storing it an object of type 
```
dosage: {
    "value": <value>,
}
```

This not only minimise the bytes but also keeps the developer more into sync to the real value type (number), so that any unintended mistake can be avoided in the future.

Reply:
Great to know if it doesn't affect other places, thanks.